### ISSUE TITLE
PublishDiagnostics issue

### ISSUE THREAD

**Issue by charlesxsh** [rust-analyzer/lsp-server#46]
I am sending a diagnostic to the VSCode with following LSP server:

```
let param = PublishDiagnosticsParams {...}
let not = lsp_server::Notification::new(N::METHOD.to_string(), params);
connection.sender.send<...>(not.into())
```

However, nothing happened in the VSCode, the notification sent successfully from the sender.send call. Can anyone enlighten me where could be the potential issue?

(the file path, start and end position is correct)

**Comment by charlesxsh** [comment]
confirmed from language client's trace, no notification received (it does receive response, but no notification received). LSP server did not report any error.

**Comment by charlesxsh** [comment]
I tried to send showmessage notification, same behavior, no error in server but nothing indicate that client receive the notification.

**Comment by charlesxsh** [comment]
Ok I figured out the reason...since lsp server is talking to the lsp client by stdio, it consumes the stdout. Make sure you don't use std print..

**Comment by lnicola** [comment]
Yeah, not sure we can do anything about that.
